### PR TITLE
Email Verification UI

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details
In this PR we're building macOS counterpart of [Simplenote iOS PR 1081](https://github.com/Automattic/simplenote-ios/pull/1081).

@eshurakov Apologies it goes beyond the 500 LOC limit, the changes required to stylize NSButton alone add up to ~90-ish LOC 🤦 

Thanks in advance!! 🥳 

Ref. #800

### Test: Basic UI
1. Log into your account and exit Simplenote
2. Checkout PR 5e5e3e3 (Contains testing code!)
3. Relaunch the App

- [x] Verify that the new Verification UI shows up after 2 seconds
- [x] Verify that the new UI looks great when you switch from/to Dark/Light mode
- [x] Verify that pressing **ESC** dismisses the new Alert

### Test: Email Update
Repeat steps 1-3 from the previous test

- [x] Verify that pressing **Change Account Email** opens the Settings UI on the web
- [x] Verify the new UI gets dismissed

### Test: Confirm Flow
Repeat steps 1-3 from the first test

- [x] Verify that pressing **Confirm** causes a Spinner to show up, and locks the screen
- [x] Verify that after a few seconds the UI switches to **Email Verification** mode
- [x] Verify that pressing Resend causes the spinner to show up again

### Release
`RELEASE-NOTES.txt` was updated in e223b42 with:

> New account verification screen

### Screenshots
Dark Light

### CODE REVIEW

File: Simplenote/AccountVerificationViewController.swift [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+        SPTracker.trackVerificationConfirmButtonTapped()
+    }
+
+    @IBAction
+    func secondaryButtonWasPressed(_ sender: Any) {
+        switch configuration {
+        case .review:
+            presentChangeEmailURL()
+            dismissWasPressed(sender)
```
Comment (eshurakov) [inline Simplenote/AccountVerificationViewController.swift]: Calling `dismissWasPressed` triggers `SPTracker.trackVerificationDismissed`, but I believe we want to send this event only when user explicitly closes this screen.

**Review by eshurakov — APPROVED** [review]
Looks great!

Only one small comments about tracks, but otherwise :shipit:

File: Simplenote/AccountVerificationViewController.swift (hunk 1 above)
Comment (jleandroperez) [reply Simplenote/AccountVerificationViewController.swift]: Ahhh nice catch!! thanks for spotting that!

**Comment by jleandroperez** [comment]
Thank you Evgeny!!

**Comment by pachlava** [comment]
[testing] All mentioned scenarios verified, checked with mbp16 | Big Sur 11.1 | SN 2.7

**Pull request merged**